starfish/image/_stack.py

```diff
@@ -104,19 +104,17 @@ def tile_opener(toc_path, tile, ext):
             pretty=True,
             tile_opener=tile_opener)
 
-    def max_proj(self, dim):
-        valid_dims = [Keys.HYB, Keys.CH, Keys.ZLAYER]
-        if dim not in valid_dims:
-            msg = "Dimension: {} not supported. Expecting one of: {}".format(dim, valid_dims)
-            raise ValueError(msg)
-
-        if dim == Keys.HYB:
-            res = numpy.max(self._data, axis=0)
-        elif dim == Keys.CH:
-            res = numpy.max(self._data, axis=1)
-        elif dim == Keys.ZLAYER and len(self._tile_shape) > 2:
-            res = numpy.max(self._data, axis=4)
-        else:
-            res = self._data
-
-        return res
+    def max_proj(self, *dims):
+        valid_dims = {
+            Keys.HYB: 0,
+            Keys.CH: 1,
+            Keys.ZLAYER: 2,
```

Comment:
rename keys.ZLAYER -> keys.Z to maintain consistency with spec.

Reply:
I would change the spec in that case. :)  Z could be misconstrued as a coordinate.

Reply:
This skimage spec seems to say, for video, that we should be using `(t, pln, row, col, ch)` http://scikit-image.org/docs/dev/user_guide/numpy_images.html

I would interpret our mapping to this as: 
```
t = hyb
pln = z
row = y
col = x
ch = ch
```
This would suggest we move `ch` to the end. I'm not sure I completely understand why they suggest this.

Reply:
Actually, let's have the naming discussion on #130. I sent out the PRs in the wrong order. My bad.